test/integration/discovery.go

```diff
@@ -147,6 +153,124 @@ var _ = Describe("Discover Cluster", func() {
 		})
 	})
 
+	Describe("Discovering affects of node with unknown state", func() {
+		var entities []*proto.EntityDTO = nil
+		var groups []*proto.GroupDTO = nil
+		var deps *appsv1.Deployment = nil
+		testName := "discovery-integration-test"
+		nodeName := "kind-worker3"
+
+		BeforeEach(func() {
+			if f.Kubeconfig.CurrentContext != "kind-kind" {
+				Skip("Skipping cluster that isn't using context kind-kind")
+			}
+
+			// create a pod to test and attach to the node soon to be in a NotReady state
+			if deps == nil {
+				var err error = nil
+				deps, err = createDeployResource(kubeClient, depSingleContainerWithResources(namespace, "", 1, false, false, false, nodeName))
+				framework.ExpectNoError(err, "Error creating test resources")
+			}
+			// stop running node worker to simulate node in NotReady state
+			execute("docker", "stop", nodeName)
```

Comment:
I would prefer to have the test fail in case of the exec error rather then simply log the error.

Reply:
> Thanks for doing this @cmedina003. Few nits and 1 timing improvement. Timing improvement is important as the overall test run time is quite considerable already.

Thank you for the review 😁 on my first kubeturbo MR!

Reply:
Tried doing it in the describe, but it was causing failures, so I wrapped the setup and reset portions of the block in `It` functions before and after all the other unknown node tests.